Feature/int 47 tb networkdata

milan-zezelj-greenstate opened the pull request.

milan-zezelj-greenstate committed:
Added fetch products request

milan-zezelj-greenstate committed:
Merge branch 'feature/int-45-tb-riverpod' into feature/int-47-tb-networkdata

milan-zezelj-greenstate committed:
Added fetching of product details

milan-zezelj-greenstate committed:
Removed unwanted import in api

milan-zezelj-greenstate committed:
Merge branch 'feature/int-45-tb-riverpod' into feature/int-47-tb-networkdata

milan-zezelj-greenstate committed:
Added product editting endpoint

milan-zezelj-greenstate committed:
Code cleanup

damjan-pantic-metaswiss commented on lib/models/product_model.dart:
```diff
@@ -1,74 +1,51 @@
-      : name = json["name"],
-        company = json["company"],
-        category = json["category"],
+      : id = json["id"],
```
use json_serializable for to and from JSON.

damjan-pantic-metaswiss commented on lib/models/product_model.dart:
```diff
@@ -1,74 +1,51 @@
+  String brand;
+  String category;
+  String thumbnail;
+  late List<String> images;
```
make all params final.

milan-zezelj-greenstate committed:
Added category fetch and did some code cleanup

milan-zezelj-greenstate committed:
Update .gitignore

milan-zezelj-greenstate committed:
Removed *.g.dart files

milan-zezelj-greenstate committed:
Added edit product provider to hande edit loading state

milan-zezelj-greenstate committed:
Added state change on list and product fetch

milan-zezelj-greenstate committed:
Fixed all TODOs and patched with changes from other PR

damjan-pantic-metaswiss commented on lib/providers/product_category_provider.dart:
```diff
@@ -0,0 +1,22 @@
+      for (final category in categoriesRaw)
+        TBPickerListItemModel(name: category)
+    ];
+    return Future.value(categories);
```
same, wrap with try catch

damjan-pantic-metaswiss reviewed (changes requested):
Merge after changes

milan-zezelj-greenstate committed:
Fixed PR comments

Now using json_serializable for converting json to model and model to json, all params are now final in the model and added try catch to category provider fetch method

milan-zezelj-greenstate committed:
Merge branch 'main' into feature/int-47-tb-networkdata
